src/Data/NodeResolver.php

```diff
@@ -582,6 +588,30 @@ public function parse_request( string $uri, $extra_query_vars = '' ) {
 		return $uri;
 	}
 
+	/**
+	 * Given a WP Post or post ID, this method attempts to resolve a preview post ID.
+	 * 
+	 * @param Int|WP_Post $post The WP Post object or Post ID
+	 * 
+	 * @return Int A preview post ID if one exists, the current post ID if one doesn't exist.
+	 */
+	public static function resolve_node_preview( $post ) {
```

Comment:
I'm not sure if this `static` method belongs here. Is there a better spot for this util? cc @jasonbahl @justlevine